libpromises/cf3lex.l

```diff
@@ -298,6 +377,22 @@ promise_type   [a-zA-Z_]+:
                           return yytext[0];
                       }
 
+<if_ignore_state><<EOF>>     {
+                               if (P.if_depth > 0)
+                               {
+                                 yyerror("EOF seen while @if was waiting for @endif");
+                                 exit(1);
+                               }
+                             }
+
+<<EOF>>     {
+              if (P.if_depth > 0)
+              {
+                yyerror("EOF seen while @if was waiting for @endif");
+                exit(1);
+              }
+              return 0; // loops forever without this
+            }
```

Comment:
When you say "loops forever without this", do you mean that the section ``<<EOF>>`` loops forever without ``return 0``, or the parser loops forever without an ``<<EOF>>`` section? If it's the former, then you can remove the whole section, since ``P.if_depth > 0`` can't be true while ``<if_ignore_state>`` is false, so all you're doing is returning 0.